Pipfile

```diff
@@ -10,11 +10,13 @@ pyopenssl = "~= 18.0"
 # Pin certbot and acme to the system-wide versions (see SE-2197)
 certbot = "==0.31.0"
 acme = "==0.31.0"
+mock = "==3.0.5"
 
 [dev-packages]
 pylint = "*"
 rope = "*"
 pytest = "*"
 
 [requires]
-python_version = "3"
+# Match version on Ubuntu LTS 16.04.6
+python_version = "3.5.2"
```

Comment:
@tartansandal I'm not very familiar with Pipfile - will this cause issues if using with python 3.5.3 or later patch versions?

Reply:
Generally not, since patch releases are supposed to be backwards compatible. The problem we are experiencing is a forwards incompatibility. The python version pinning does not force us to use that version, just that all the other packages in the virtual env are assuming we are using that version. Pipenv is crackrock IMHO: I'm trying to prevent it from randomly upgrading to untested configurations.